app/javascript/controllers/calendar_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["date"]
+
+  connect() {
```

Comment:
connect happens with each monthly calendar that is created in the DOM. so this cycles through each date and adds a toggle to each date. the `date` is the hidden field, and the parent is the td box itself, which is the interactable element in the UI.